linkerd/app/outbound/src/lib.rs

```diff
@@ -101,67 +100,25 @@ impl Config {
         C: tower::Service<TcpEndpoint, Error = Error> + Unpin + Clone + Send + Sync + 'static,
         C::Response: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
         C::Future: Unpin + Send,
-        E: Resolve<net::SocketAddr, Endpoint = Metadata, Error = Error>
-            + Unpin
-            + Clone
-            + Send
-            + 'static,
-        E::Future: Unpin + Send,
-        E::Resolution: Unpin + Send,
+        R: Resolve<Addr, Endpoint = Metadata, Error = Error> + Unpin + Clone + Send + 'static,
+        R::Future: Unpin + Send,
+        R::Resolution: Unpin + Send,
         I: tokio::io::AsyncRead + tokio::io::AsyncWrite + std::fmt::Debug + Unpin + Send + 'static,
     {
-        let watchdog = self.proxy.cache_max_idle_age * 2;
```

Comment:
nit/tioli: seems like we could just stick a `fn watchdog_timeout` on `Config` or on `Proxy`?

Reply:
will address this in a followup cleanup